### ISSUE TITLE
Display ETA and details when starting a workspace

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#3738]
Today, there are very little details, on what's happening, or what should the user wait for, while a workspace is being prepared.

I read some interesting explanations, `Workspace Life-Cycles` on this page -
https://www.notion.so/Redesign-Loading-Screen-5e2f3a45cb5446088566af1b30cde55f

I think it will be a nicer experience, to know what to expect, when that is possible.
If you think it's overwhelming amount of information, It could be minimized by default, and the user can open for details.

During a prebuild, a user can at least see the commands that are running, so even that ETA is unknown, the user can tell where the process is at according to what is running in the terminal preview screen.

The other steps, like `Acquiring Node`, and `Pulling Docker Images`, can take more than a few seconds, so having details like ETA, average time for that task, and a count of MB downloaded out of total image MB - would make waiting a better experience, because it's no longer unknown, or feel like it's stuck, the user would know more about what is happening.

**Comment by svenefftinge** [comment]
Yes, showing an ETA would be great. We don't have that information ATM, unfortunately.

**Comment by shaal** [comment]
Requesting to add `meta: never-stale` label to this issue

**Comment by gtsiolis** [comment]
Added never-stale label although providing an ETA still sounds hard to tackle and potentially has a low impact for the users.

Instead focusing on improving the a) user experience and b) perceived performance of the loading screen while c) minimizing the start-up time sounds more realistic and impactful. ↪️

**Comment by shaal** [comment]
Thank you @gtsiolis.
ETA would be great, but if it's not possible to know the exact numbers, I would suggest a visual ETA.

Currently, when waiting for workspace to start, Gitpod displays 3 bars, with 1 written status line above it. There's no way to know if it's going to take 5 seconds or 30.

I think the display, when starting a workspace can be improved by:
* Display the list of all steps that are going to happen (I think displaying any step that takes over 3 seconds would be nice)
* Display a checkmark for any step that was completed.
* For each step display a bar filling up
The rate of how fast a bar gets filled up would be determined by the worst-case-scenario estimation. That way, when a step _might_ take 30 seconds, the bar start filling up, and even if it's much shorter, like 5 seconds, the user will be happy to see a step completed, and the new step started.

A nice addition to that, is if a certain step takes MORE than the worst-case-scenario estimated, the user would get a friendly message that something is not working correctly, with suggested steps on how to get unblocked (refresh? try again? submit a ticket)

**Comment by atduarte** [comment]
Another benefit of changing the loading screen could be that we make the distinction between when a prebuild is used and when it's not clearer. 

Example interface (from Vercel) with process steps, time, and some associated logs:

| Default  | Expanded |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/666176/131521440-8c3aac1b-3743-4104-b6af-5575cdb76343.png) | ![image](https://user-images.githubusercontent.com/666176/131521471-9644185c-7c1e-428e-bfa0-48e53eb93590.png) |

**Comment by JohannesLandgraf** [comment]
@shaal and myself did a demo today and received the feedback that showing more information here / being more transparent would help a lot

**Comment by loujaybee** [comment]
I think this is related, or a duplicate of: 
- https://github.com/gitpod-io/gitpod/issues/11199

The linked issue is a bit more specific, to define these events, but this issue looks like it's about having them viewable on workspace start. The two are certainly related and/or solving similar challenges.
